app/api/sessions.py

```diff
@@ -130,6 +130,17 @@ def before_get_object(self, view_kwargs):
             event = safe_query(self, Event, 'identifier', view_kwargs['event_identifier'], 'identifier')
             view_kwargs['event_id'] = event.id
 
+    def before_update_object(self, session, data, view_kwargs):
+        """
+        before update method to verify if session is locked before updating session object
+        :param event:
+        :param data:
+        :param view_kwargs:
+        :return:
+        """
+        if session.is_locked and data.get('is_locked') == session.is_locked:
```

Comment:
I am a bit sceptical about this condition, does an already unlocked session gets locked after this?

Reply:
No. This condition is to raise error if a locked session is edited anyhow.
I added this condition `data.get('is_locked') == session.is_locked` to `session.is_locked` because if I used `session.is_locked` alone, error was raised even when I changed state of session's is_locked, which makes sense.

Reply:
@shreyanshdwivedi Okay, makes sense. But wouldn't toggling the lock button go through this and hence raise a forbidden error upon unlocking?

Reply:
That's why I used `data.get('is_locked') == session.is_locked` because when we'll be toggling the is_locked property, this condition will never be satisfied @uds5501

Reply:
@shreyanshdwivedi Understood.

Reply:
If `data.get('is_locked')` is None, this will never be true and error won't be raised. Logic is wrong

Also, what's the point of locking a session if you can unlock it and then edit. Shouldn't the unlocking only be allowed for an admin?